src/java/org/apache/cassandra/repair/autorepair/RepairAssignmentIterator.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.repair.autorepair;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * Convenience {@link Iterator} implementation for to assist implementations of
+ * {@link IAutoRepairTokenRangeSplitter#getRepairAssignments(boolean, List)} by passing {@link KeyspaceRepairPlan}
+ * to a custom {@link #nextInternal(int, KeyspaceRepairPlan)} in priority order.
+ */
+public abstract class RepairAssignmentIterator implements Iterator<KeyspaceRepairAssignments>
+{
+    private final Iterator<PrioritizedRepairPlan> repairPlanIterator;
+
+    private Iterator<KeyspaceRepairPlan> currentIterator = null;
+    private PrioritizedRepairPlan currentPlan = null;
+
+    RepairAssignmentIterator(List<PrioritizedRepairPlan> repairPlans)
```

Comment:
this is a public class, but the constructor is only package private, maybe this should be protected instead? so other implementations can extend without needing to hack into the `org.apache.cassandra.repair.autorepair` package.

Reply:
Ah, this is an oversight.  I wanted for this to be public so custom implementations can use this, as it cuts down some boilerplate on producing an `Iterator<KeyspaceRepairAssignments>`.  Will make this public.